src/App.tsx

```diff
@@ -1,28 +1,131 @@
+import { useState } from 'react';
 import './App.scss';
+import cn from 'classnames';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+type Todo = {
+  id: number,
+  title: string,
+  completed: boolean,
+  userId: number,
+};
+
+const todosFormServer: Todo[] = [...todosFromServer];
 
 export const App = () => {
+  const [titleInput, setTitleInput] = useState('');
+  const [userSelected, setUserSelected] = useState(0);
+  const [newTodos, setNewTodos] = useState(todosFormServer);
+  const [errorMassegeTitle, setErrorMassegeTitle] = useState(false);
+  const [errorMassegeUser, setErrorMassegeUser] = useState(false);
+
+  function createTodo(
+    id: number, title: string, completed: boolean, userId: number,
+  ) {
+    const newTodo = {
+      id,
+      title,
+      completed,
+      userId,
+    };
+
+    setNewTodos([...newTodos, newTodo]);
+  }
+
   return (
     <div className="App">
       <h1>Add todo form</h1>
 
-      <form action="/api/users" method="POST">
+      <form
+        action="/api/users"
+        method="POST"
+        onSubmit={(event) => {
+          event.preventDefault();
+          if (titleInput === '') {
+            setErrorMassegeTitle(true);
+          } else {
+            setErrorMassegeTitle(false);
+          }
+
+          if (!userSelected) {
+            setErrorMassegeUser(true);
+          } else {
+            setErrorMassegeUser(false);
+          }
+
+          if (titleInput && userSelected) {
+            createTodo(
+              newTodos.length + 1,
+              titleInput,
+              false,
+              userSelected,
+            );
+            setTitleInput('');
+            setUserSelected(0);
+          }
+        }}
+      >
         <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
+          <label htmlFor="input">
+            <span>Title: </span>
+          </label>
+
+          <input
+            type="text"
+            id="input"
+            data-cy="titleInput"
+            value={titleInput}
+            placeholder="Enter a title"
+            onChange={({ target }) => {
```

Comment:
it is bad practice to make arrow anonymous function in JSX
make handler in component and pass reference here.
check other similar cases